File: DynamicProgramming1/FibonacciFunction/PythonFibonacciFunction/FibonacciFunction.py
import sys


MAX = 41
memo_0 = [1, 0]
memo_1 = [0, 1]


def fibonacci_function():
    for i in range(2, MAX):
        memo_0.append(memo_0[i - 1] + memo_0[i - 2])
        memo_1.append(memo_1[i - 1] + memo_1[i - 2])


fibonacci_function()

t = int(sys.stdin.readline())

for _ in range(t):
    n = int(sys.stdin.readline())
    print(memo_0[n], memo_1[n])

File: DynamicProgramming1/FibonacciFunction/CSFibonacciFunction/FibonacciFunction.cs
using System;
using System.Collections.Generic;

namespace CSFibonacciFunction
{
    static class Constants
    {
        public const int Max = 41;
    }

    class FibonacciFunction
    {
        static List<int> memo0 = new List<int>(Constants.Max);
        static List<int> memo1 = new List<int>(Constants.Max);

        static void FibFunc()
        {
            memo0.Add(1);
            memo1.Add(0);
            memo0.Add(0);
            memo1.Add(1);

            for (int i = 2; i < Constants.Max; i++)
            {
                memo0.Add(memo0[i - 1] + memo0[i - 2]);
                memo1.Add(memo1[i - 1] + memo1[i - 2]);
            }
        }

        static void Main(string[] args)
        {
            FibFunc();

            int t = Convert.ToInt32(Console.ReadLine());

            for (int i = 0; i < t; i++)
            {
                int n = Convert.ToInt32(Console.ReadLine());
                Console.WriteLine("{0} {1}", memo0[n], memo1[n]);
            }
        }
    }
}
